server/parsec/client_context.py

```diff
@@ -176,9 +176,9 @@ class AnonymousAccountClientContext:
     settled_api_version: ApiVersion
     logger: ParsecBoundLogger = field(init=False)
     client_user_agent: str
-    # there is no guarantee that asgi is able to provide the ip address
+    # In ASGI the client IP address is optional
     # see https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope
-    client_ip: str | None  # TODO replace by ipaddr #10384
+    client_ip_address: str | None
```

Comment:
nit: Is there a strong reason to keep the None variant ? In VaultItemRecoveryAuthMethod the string can be empty

edit: and we are not checking the ip format

Reply:
The conversion `None` -> `""` can take place at multiple levels:
1. when client_context is created
2. when the data is stored (e.g. in `MemoryAccountComponent.create_account`)
3. when the data is displayed (e.g. in `MemoryAccountComponent.vault_item_recovery_list`)

I was reluctant to do that early on in step 1 since it means "losing information" (i.e. the database contains cooked data, which is generally not a good idea since the need for cooking those data may change with time, so better store raw data and only do the cooking when fetching them)

However I think you are right here: if the cooking had an actual impact then we wouldn't consider `None` is equivalent to `""` in the first place !
So I'm changing this code to do the cooking in step 1 :+1: